vividus-http-client/src/main/java/org/vividus/http/client/HttpRequestRetryStrategy.java

```diff
@@ -14,17 +14,26 @@
  * limitations under the License.
  */
 
-package org.vividus.http.handler;
+package org.vividus.http.client;
 
+import java.time.Duration;
 import java.util.List;
+import java.util.Optional;
+import java.util.Set;
 
-import org.apache.http.HttpRequest;
-import org.apache.http.impl.client.DefaultHttpRequestRetryHandler;
+import org.apache.hc.client5.http.impl.DefaultHttpRequestRetryStrategy;
+import org.apache.hc.core5.http.HttpRequest;
+import org.apache.hc.core5.util.TimeValue;
 
-public class IdempotentMethodsRetryHandler extends DefaultHttpRequestRetryHandler
+public class HttpRequestRetryStrategy extends DefaultHttpRequestRetryStrategy
```

Comment:
should not we have Idempotent as part of class name?

Reply:
no, because it includes idempotent requests handling and old service unavailable retry strategy